DOC: Document abi3 status

larsoner opened the pull request:
PR Checklist:

- [x] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] ~~if you are adding a new page under `docs/` or `community/`, you have added it to the sidebar in the corresponding `_sidebar.json` file~~
- [x] put any other relevant information below

I maintain a few packages whose wheels are abi3, it would be nice to be able to use them. But I think it's currently not possible, so want to help the next person know that if they ctrl-F "abi3" on the knowledge base page, and know what issue to subscribe to. Suggestions welcome if this doesn't accurately describe the current status or what people should subscribe to for status updates!

Closes #1865

larsoner commented:
:+1: added a link to that as well

wolfv commented:
@larsoner in case you are not aware, `rattler-build` has shipped CEP-20 support and you can use it on conda-forge :) Here is an example: https://github.com/conda-forge/py-rattler-feedstock/blob/main/recipe/recipe.yaml

larsoner commented:
@wolfv @isuruf here's the doc update from https://github.com/conda-forge/python-feedstock/pull/669, let me know if it needs tweaks! I left out `rattler-build` stuff since I wanted to stick to vanilla `conda-build` here but a follow-up PR could add it.

larsoner commented:
[Looks good](https://deploy-preview-2421--conda-forge-previews.netlify.app/docs/maintainer/knowledge_base/#abi3) to me :smile: (blurred out `noarch` section just for emphasis on new one, big screenshot so you can see the section hierarchy / placement):

![Screenshot From 2025-04-18 10-59-57](https://github.com/user-attachments/assets/c613fb31-4db6-40d9-9279-9ab638dd8ec8)

FYI I changed it from a subsection of the `Noarch` section and changed it to a subsection of the `Special Dependencies and Packages` section, as it seemed more appropriate.
